userchroot.c

```diff
@@ -682,6 +682,46 @@ int main(int argc, char* argv[], char* envp[]) {
         }
       }
 
+      // Cleanup the /dev/shm directory created by child_fn
+      int shm_path_len = strlen(final_path) + strlen("/dev/shm") + 1;
+      char* shm_path = malloc(shm_path_len);
+      if (shm_path == NULL) {
+        fprintf(stderr,"Failed to allocate memory. Aborting.\n");
+        exit(ERR_EXIT_CODE);
+      }
+      rc = snprintf(shm_path, shm_path_len, "%s/%s", final_path, "dev/shm");
+      if (rc <= 0) {
+        fprintf(stderr,"Failed to assemble path. Aborting.\n");
+        exit(ERR_EXIT_CODE);
+      }
+      rc = rmdir(shm_path);
+      if (rc) {
+        fprintf(stderr, "Failed to rmdir %s. Aborting.\n", shm_path);
+        exit(ERR_EXIT_CODE);
```

Comment:
`rmdir()` may fail if the chroot directory is still used by another command (another userchroot session has `/dev/shm` still mounted in its mount namespace). In the spirit of #12 and #17 I wouldn't immediately abort here. If multiple concurrent userchroot sessions in the same chroot directory are considered supported, we should probably not even print or return an error.